frontend/app_flowy/lib/workspace/presentation/home/menu/app/section/section.dart

```diff
@@ -26,29 +29,111 @@ class ViewSection extends StatelessWidget {
       child: Consumer(builder: (context, ViewSectionNotifier notifier, child) {
-        return _renderSectionItems(context, notifier.views);
+        return RenderSectionItems(views: notifier.views);
       }),
     );
   }
 
-  Widget _renderSectionItems(BuildContext context, List<View> views) {
-    List<Widget> viewWidgets = [];
-    if (views.isNotEmpty) {
-      viewWidgets = views
-          .map(
-            (view) => ViewSectionItem(
-              view: view,
-              isSelected: _isViewSelected(context, view.id),
-              onSelected: (view) {
-                context.read<ViewSectionNotifier>().selectedView = view;
-                Provider.of<MenuSharedState>(context, listen: false).selectedView.value = view;
-              },
-            ).padding(vertical: 4),
-          )
-          .toList(growable: false);
+  // Widget _renderSectionItems(BuildContext context, List<View> views) {
+  //   List<Widget> viewWidgets = [];
+  //   if (views.isNotEmpty) {
+  //     viewWidgets = views
+  //         .map(
+  //           (view) => ViewSectionItem(
+  //             view: view,
+  //             isSelected: _isViewSelected(context, view.id),
+  //             onSelected: (view) {
+  //               context.read<ViewSectionNotifier>().selectedView = view;
+  //               Provider.of<MenuSharedState>(context, listen: false).selectedView.value = view;
+  //             },
+  //           ).padding(vertical: 4),
+  //         )
+  //         .toList(growable: false);
+  //   }
+
+  //   return Column(children: viewWidgets);
+  // }
+
+  // bool _isViewSelected(BuildContext context, String viewId) {
+  //   final view = context.read<ViewSectionNotifier>().selectedView;
+  //   if (view == null) {
+  //     return false;
+  //   }
+  //   return view.id == viewId;
+  // }
+}
+
+class RenderSectionItems extends StatefulWidget {
+  const RenderSectionItems({Key? key, required this.views}) : super(key: key);
+
+  final List<View> views;
+
+  @override
+  State<RenderSectionItems> createState() => _RenderSectionItemsState();
+}
+
+class _RenderSectionItemsState extends State<RenderSectionItems> {
+  List<View> views = <View>[];
+
+  /// Maps the hasmap value of the section items to their index in the reorderable list.
+  //TODO @gaganyadav80: Retain this map to persist the order of the items.
+  final Map<String, int> _sectionItemIndex = <String, int>{};
+
+  void _initItemList() {
+    views.addAll(widget.views);
+
+    for (int i = 0; i < views.length; i++) {
+      if (_sectionItemIndex[views[i].id] == null) {
+        _sectionItemIndex[views[i].id] = i;
+      }
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _initItemList();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (views.isEmpty) {
+      _initItemList();
     }
 
-    return Column(children: viewWidgets);
+    log("BUILD: Section items: ${views.length}");
```

Comment:
You could use Log.info

Reply:
I forgot to remove this but if I use this in future then I will keep it in mind. :)